fe/fe-core/src/main/java/org/apache/doris/analysis/AlterColumnStatsStmt.java

```diff
@@ -47,59 +60,124 @@ public class AlterColumnStatsStmt extends DdlStmt {
+        checkPartitionAndColumnNames();
+
         // check properties
         Optional<StatsType> optional = properties.keySet().stream().map(StatsType::fromString)
-                .filter(statsType -> !CONFIGURABLE_PROPERTIES_SET.contains(statsType)).findFirst();
+                .filter(statsType -> !CONFIGURABLE_PROPERTIES_SET.contains(statsType))
+                .findFirst();
         if (optional.isPresent()) {
-            throw new AnalysisException(optional.get() + " is invalid statistic");
+            throw new AnalysisException(optional.get() + " is invalid statistics");
         }
+
         // check auth
-        if (!Env.getCurrentEnv().getAuth().checkTblPriv(
-                ConnectContext.get(), tableName.getDb(), tableName.getTbl(), PrivPredicate.ALTER)) {
+        if (!Env.getCurrentEnv().getAuth()
+                .checkTblPriv(ConnectContext.get(), tableName.getDb(), tableName.getTbl(), PrivPredicate.ALTER)) {
             ErrorReport.reportAnalysisException(ErrorCode.ERR_TABLEACCESS_DENIED_ERROR, "ALTER COLUMN STATS",
-                    ConnectContext.get().getQualifiedUser(),
-                    ConnectContext.get().getRemoteIP(),
+                    ConnectContext.get().getQualifiedUser(), ConnectContext.get().getRemoteIP(),
                     tableName.getDb() + ": " + tableName.getTbl());
         }
+
         // get statsTypeToValue
         properties.forEach((key, value) -> {
             StatsType statsType = StatsType.fromString(key);
             statsTypeToValue.put(statsType, value);
         });
     }
 
-    public TableName getTableName() {
-        return tableName;
-    }
+    private void checkPartitionAndColumnNames() throws AnalysisException {
+        Database db = analyzer.getEnv().getInternalDataSource()
+                .getDbOrAnalysisException(tableName.getDb());
+        Table table = db.getTableOrAnalysisException(tableName.getTbl());
 
-    public String getColumnName() {
-        return columnName;
-    }
+        if (table.getType() != Table.TableType.OLAP) {
+            throw new AnalysisException("Only OLAP table statistics are supported");
+        }
 
-    public Map<StatsType, String> getStatsTypeToValue() {
-        return statsTypeToValue;
+        OlapTable olapTable = (OlapTable) table;
+
+        if (olapTable.getColumn(columnName) == null) {
+            ErrorReport.reportAnalysisException(ErrorCode.ERR_WRONG_COLUMN_NAME, columnName);
+        }
+
+        if (optPartitionNames != null) {
+            optPartitionNames.analyze(analyzer);
+            if (!olapTable.isPartitioned()) {
+                throw new AnalysisException("Not a partitioned table: " + olapTable.getName());
+            }
+            Set<String> olapPartitionNames = olapTable.getPartitionNames();
+            Optional<String> optional = optPartitionNames.getPartitionNames().stream()
+                    .filter(name -> !olapPartitionNames.contains(name)).findFirst();
```

Comment:
it is better to use anyMatch instead of findFirst and isPresent.

Reply:
indeed anyMatch looks better, but I use findFirst here mainly to tell the user the name of the partition that doesn't exist.